geode-core/src/main/java/org/apache/geode/management/internal/cli/domain/RegionConfigFactory.java

```diff
@@ -32,310 +32,255 @@
+      regionAttributesType.setKeyConstraint(args.getKeyConstraint());
     }
 
     if (args.getValueConstraint() != null) {
-      addAttribute(regionConfig, a -> a.setValueConstraint(args.getValueConstraint()));
+      regionAttributesType.setValueConstraint(args.getValueConstraint());
     }
 
     if (args.getStatisticsEnabled() != null) {
-      addAttribute(regionConfig, a -> a.setStatisticsEnabled(args.getStatisticsEnabled()));
+      regionAttributesType.setStatisticsEnabled(args.getStatisticsEnabled());
     } else if (regionAttributes != null) {
-      addAttribute(regionConfig, a -> a.setStatisticsEnabled(regionAttributes
-          .getStatisticsEnabled()));
+      regionAttributesType.setStatisticsEnabled(regionAttributes.getStatisticsEnabled());
     }
 
-    if (args.getEntryExpirationIdleTime() != null) {
-      RegionAttributesType.EntryIdleTime entryIdleTime = new RegionAttributesType.EntryIdleTime();
-      entryIdleTime.setExpirationAttributes(
-          args.getEntryExpirationIdleTime().getExpirationAttributes().toConfigType());
-      addAttribute(regionConfig, a -> a.setEntryIdleTime(entryIdleTime));
-    } else if (regionAttributes != null &&
-        regionAttributes.getEntryIdleTimeout() != null &&
-        !regionAttributes.getEntryIdleTimeout().isDefault()) {
-      RegionAttributesType.EntryIdleTime entryIdleTime = new RegionAttributesType.EntryIdleTime();
-      entryIdleTime.setExpirationAttributes(regionAttributes
-          .getEntryIdleTimeout().toConfigType());
-      addAttribute(regionConfig, a -> a.setEntryIdleTime(entryIdleTime));
-    }
-
-    if (args.getEntryIdleTimeCustomExpiry() != null) {
-      Object maybeEntryIdleAttr = getAttribute(regionConfig, a -> a.getEntryIdleTime());
-      RegionAttributesType.EntryIdleTime entryIdleTime =
-          maybeEntryIdleAttr != null ? (RegionAttributesType.EntryIdleTime) maybeEntryIdleAttr
-              : new RegionAttributesType.EntryIdleTime();
-
-      ExpirationAttributesType expirationAttributes;
-      if (entryIdleTime.getExpirationAttributes() == null) {
-        expirationAttributes = new ExpirationAttributesType();
-        expirationAttributes.setTimeout("0");
-      } else {
-        expirationAttributes = entryIdleTime.getExpirationAttributes();
+    // first get the expiration attributes from the command options
+    regionAttributesType.setEntryIdleTime(getExpirationAttributes(args.getEntryExpirationIdleTime(),
+        args.getEntryIdleTimeCustomExpiry()));
+    regionAttributesType.setEntryTimeToLive(
+        getExpirationAttributes(args.getEntryExpirationTTL(), args.getEntryTTLCustomExpiry()));
+    regionAttributesType
+        .setRegionIdleTime(getExpirationAttributes(args.getRegionExpirationIdleTime(), null));
+    regionAttributesType
+        .setRegionTimeToLive(getExpirationAttributes(args.getRegionExpirationTTL(), null));
+
+    // if regionAttributes has these attributes, then use that
```

Comment:
Maybe this is more stylistic, but you've probably noticed that coming from the Go-style I prefer ifs to be flat and not hierarchical, as much as possible. I'm ok with either way, but I've noticed that in this class we now have a dichotomy, i.e. we're doing both flat ifs and hierarchical ifs when it comes to `if (regionAttributes != null)`. Can we be consistent across this class? I.e. either flatten them all out or group them all into a global `if (regionAttributes != null)`?

Reply:
I think I grouped these four into one global `if (regionAttributes != null)` because they all deal with expiration attributes

Reply:
Right -- but I'm wondering if it's worth just grouping all of them into one global `if (regionAttributes != null)` in order to be consistent.